### PR TITLE
DNM: Add predefined identifier __func__ to dout and modify log messages

### PR DESCRIPTION
**Pull request by shinobu-x**
mon/OSDMonitor:: Add ``__func__`` macro to ``dout`` because some messages have already ``dout`` but some don't.

Knowing function name could help sysadmins do some stuff in case that they face problem(s).

Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -368,8 +368,8 @@ void OSDMonitor::update_from_paxos(bool *need_bootstrap)
     assert(err == 0);
     assert(inc_bl.length());
 
-    dout(7) << "update_from_paxos  applying incremental " << osdmap.epoch+1
-	    << dendl;
+    dout(7) << __func__ << "update_from_paxos applying incremental "
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: s/update_from_paxos// since we print `__func__` already.

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -558,7 +561,7 @@ void OSDMonitor::on_shutdown()
 
 void OSDMonitor::update_logger()
 {
-  dout(10) << "update_logger" << dendl;
+  dout(10) << __func__ << " update_logger" << dendl;
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: ditto, remove " update_logger" please.

File: src/mon/OSDMonitor.cc [hunk 3]
```diff
@@ -571,11 +574,11 @@ void OSDMonitor::create_pending()
   pending_inc = OSDMap::Incremental(osdmap.epoch+1);
   pending_inc.fsid = mon->monmap->fsid;
 
-  dout(10) << "create_pending e " << pending_inc.epoch << dendl;
+  dout(10) << __func__ << " create_pending e " << pending_inc.epoch << dendl;
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: ditto.

File: src/mon/OSDMonitor.cc [hunk 4]
```diff
@@ -571,11 +574,11 @@ void OSDMonitor::create_pending()
...
 
   // clean up pg_temp, primary_temp
   OSDMap::clean_temps(g_ceph_context, osdmap, &pending_inc);
-  dout(10) << "create_pending  did clean_temps" << dendl;
+  dout(10) << __func__ << " create_pending  did clean_temps" << dendl;
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: ditto.

File: src/mon/OSDMonitor.cc [hunk 5]
```diff
@@ -880,8 +882,8 @@ void OSDMonitor::prime_pg_temp(
  */
 void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
 {
-  dout(10) << "encode_pending e " << pending_inc.epoch
-	   << dendl;
+  dout(10) << __func__ << "encode_pending e " << pending_inc.epoch
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: ditto.

File: src/mon/OSDMonitor.cc [hunk 6]
```diff
@@ -915,7 +917,8 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
   auto p = pending_inc.new_state.begin();
   while (p != pending_inc.new_state.end()) {
     if (p->second == 0) {
-      dout(10) << "new_state for osd." << p->first << " is 0, removing" << dendl;
+      dout(10) << __func__ << "new_state for osd." << p->first
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: add a space before "new_state"

File: src/mon/OSDMonitor.cc [hunk 7]
```diff
@@ -1490,7 +1497,8 @@ bool OSDMonitor::prepare_update(MonOpRequestRef op)
 {
   op->mark_osdmon_event(__func__);
   Message *m = op->get_req();
-  dout(7) << "prepare_update " << *m << " from " << m->get_orig_source_inst() << dendl;
+  dout(7) << __func__ << "prepare_update " << *m << " from "
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: remove "prepare_update ".

**Comment by shinobu-x** [comment]
@tchaikov 
Thanks for your review. Please have a look when you get a chance.

File: src/mon/OSDMonitor.cc [hunk 8]
```diff
@@ -10892,7 +10930,8 @@ bool OSDMonitor::prepare_pool_op(MonOpRequestRef op)
   case POOL_OP_CREATE_SNAP:
     if (!pp.snap_exists(m->name.c_str())) {
       pp.add_snap(m->name.c_str(), ceph_clock_now());
-      dout(10) << "create snap in pool " << m->pool << " " << m->name << " seq " << pp.get_snap_epoch() << dendl;
+      dout(10) << " create snap in pool " << m->pool << " " << m->name
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: why did you add a space before "create"?

File: src/mon/OSDMonitor.cc [hunk 9]
```diff
@@ -10785,9 +10822,10 @@ bool OSDMonitor::preprocess_pool_op_create(MonOpRequestRef op)
     return true;
   }
   if (!session->is_capable("osd", MON_CAP_W)) {
-    dout(5) << "attempt to create new pool without sufficient auid privileges!"
-	    << "message: " << *m  << std::endl
-	    << "caps: " << session->caps << dendl;
+    dout(5) << __func__
+            << "attempt to create new pool without sufficient auid privileges!"
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: add a space before "attempt".

File: src/mon/OSDMonitor.cc [hunk 10]
```diff
@@ -9582,7 +9617,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
       err = -EPERM;
       goto reply;
     } else if (!osdmap.exists(id)) {
-      ss << "osd." << id << " does not exist";
+      ss << "osd." << id << " does not exist.";
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: please note this message is not sent to logging. so could you please split this change into another commit?

File: src/mon/OSDMonitor.cc [hunk 11]
```diff
@@ -8213,8 +8248,8 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
 	err = normalize_profile(name, profile_map, true, &ss);
 	if (err)
 	  goto reply;
-	dout(20) << "erasure code profile set " << profile << "="
-		 << profile_map << dendl;
+    dout(20) << __func__ << " erasure code profile set " << profile << "="
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: wrong indent.

File: src/mon/OSDMonitor.cc [hunk 12]
```diff
@@ -8200,7 +8234,8 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     if (profile == "default") {
       if (!osdmap.has_erasure_code_profile(profile)) {
 	if (pending_inc.has_erasure_code_profile(profile)) {
-	  dout(20) << "erasure code profile " << profile << " already pending" << dendl;
+      dout(20) << __func__ << " erasure code profile " << profile
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: wrong indent.

File: src/mon/OSDMonitor.cc [hunk 13]
```diff
@@ -8175,8 +8209,8 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
 	}
       }
 
-      dout(20) << "erasure code profile set " << name << "="
-	       << profile_map << dendl;
+      dout(20) << " erasure code profile set " << name << "=" << profile_map
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: why the heading space?

File: src/mon/OSDMonitor.cc [hunk 14]
```diff
@@ -8133,7 +8167,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     string plugin = profile_map["plugin"];
 
     if (pending_inc.has_erasure_code_profile(name)) {
-      dout(20) << "erasure code profile " << name << " try again" << dendl;
+      dout(20) << " erasure code profile " << name << " try again" << dendl;
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: rasure code profile " <

File: src/mon/OSDMonitor.cc [hunk 15]
```diff
@@ -5511,23 +5542,22 @@ int OSDMonitor::prepare_new_pool(string& name, uint64_t auid,
...
     }
-    dout(10) << __func__ << " crush smoke test duration: "
-             << duration << dendl;
+    dout(10) << __func__ << " crush smoke test duration: " << duration << dendl;
   }
   unsigned size, min_size;
   r = prepare_pool_size(pool_type, erasure_code_profile, &size, &min_size, ss);
   if (r) {
-    dout(10) << " prepare_pool_size returns " << r << dendl;
+    dout(10) << __func__ << " returns " << r << dendl;
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: please reread the logging message.

File: src/mon/OSDMonitor.cc [hunk 16]
```diff
@@ -5404,7 +5435,7 @@ int OSDMonitor::get_crush_rule(const string &rule_name,
     ret = newcrush.get_rule_id(rule_name);
     if (ret != -ENOENT) {
       // found it, wait for it to be proposed
-      dout(20) << __func__ << ": rule " << rule_name
+      dout(20) << __func__ << " rule " << rule_name
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: i am not sure why you are removing the ":"? is this part of the "__func__" change?

File: src/mon/OSDMonitor.cc [hunk 17]
```diff
@@ -4940,7 +4971,8 @@ int OSDMonitor::prepare_new_pool(MonOpRequestRef op)
 {
   op->mark_osdmon_event(__func__);
   MPoolOp *m = static_cast<MPoolOp*>(op->get_req());
-  dout(10) << "prepare_new_pool from " << m->get_connection() << dendl;
+  dout(10) << __func__ << "prepare_new_pool from " << m->get_connection()
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: remove "prepare_new_pool".

File: src/mon/OSDMonitor.cc [hunk 18]
```diff
@@ -3381,16 +3412,16 @@ void OSDMonitor::tick()
...
+  } else
+    dout(10) << __func__ << " NOOUT flag set, not checking down osds" << dendl;
 
   // expire blacklisted items?
   for (ceph::unordered_map<entity_addr_t,utime_t>::iterator p = osdmap.blacklist.begin();
        p != osdmap.blacklist.end();
        ++p) {
     if (p->second < now) {
-      dout(10) << "expiring blacklist item " << p->first << " expired " << p->second << " < now " << now << dendl;
+      dout(10) << __func__ << "expiring blacklist item " << p->first
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: add a space before "expiring".

**Comment by tchaikov** [comment]
also, fwiw, `__func__` is not a macro, it is a constant string variable.

**Comment by gregsfortytwo** [comment]
Closing this since it's marked DNM and has conflicts.
